Agents/PubChemAgent/README.md

```diff
@@ -0,0 +1,195 @@
+- STACK_NAME=TEST         # to be left blank for "standalone" deployment
+# Additional environment variables required for Stack deployment
+# (can be left blank for "standalone" deployment)
+- NAMESPACE=ontospecies   # Target Blazegraph namespace
+# Additional environment variables required for "standalone deployment"
+# (can be left blank for Stack deployment)
+- QUERY_ENDPOINT=
+- UPDATE_ENDPOINT=
+```
+
+The STACK_NAME variable is used to identify the deployment mode of the agent. In case the STACK_NAME is left blank, default Blazegraph endpoint setting will be taken from the docker-compose file. Otherwise they will be retrieved using the StackClients based on the provided NAMESPACE variable.
+
+Please note:
+
+All variables defined here (except for STACK_NAME) serve as default values. To omit any of those default values, either remove the key completely or just leave it blank.
+
+A missing STACK_NAME variable will result in an error; however, when deploying using the stack-manager start up script, the STACK_NAME variable will be set automatically for all services. Hence, this could be left blank here; however, if provided, it needs to match the STACK_NAME used by the stack-manager!
+
+### Build and publish the Docker image
+
+To build and publish the agent Docker image please use the following commands. Please note that all of those commands are bundled in the publish_docker_image.sh convenience script (only target image (i.e. production/debug) needs to be adjusted at top of that script).
+
+```bash
+# Building the Docker image (production / debug)
+docker-compose -f docker-compose.yml  build
+docker-compose -f docker-compose.debug.yml  build
+
+# Publish the Docker image to the Github container registry
+docker image push ghcr.io/cambridge-cares/<image tag>:<version>
+```
+
+Time out issues have been observed when building the image. If this happens, please try pulling the required stack-clients image first by
+
+```bash
+docker pull docker.cmclinnovations.com/stack-client:1.6.2
+```
+
+### Docker deployment
+
+Deploy the dockerised agent by running the following code in the command prompt from the same location where this README is located (ideally, use a bash terminal to avoid potential issues with inconsistent path separators).
+
+```bash
+# Deploy the Docker images (production / debug) locally
+docker-compose -f docker-compose.yml  up
+docker-compose -f docker-compose.debug.yml  up
+```
+
+To verify the correct startup of the agent, open the URL address the agent is running on, e.g. <http://localhost:5000> in your browser.
+
+### Stack Deployment
+
+If you want to spin up this agent as part of a stack, do the following:
+
+- Build the production image using the commands provided above (do not spin up the image)
+- Copy the pubchem-agent.json file from the stack-manager-input-config folder into the inputs/config/services folder of the stack manager
```

Comment:
You may want to mention that people may want to change the namespace in this file before spinning up the agent.

Reply:
changed as follow:

-Copy the pubchem-agent.json file from the stack-manager-input-config folder into the inputs/config/services folder of the stack manager. The enviromental variable NAMESPACE can be changed in pubchem-agent.json before spinning up the agent.